File: CS.NodeJS.LoadTest.WebApp/webapp/data/memory/eventResultMemoryDAO.js
module.exports = {

  _database: [],

  insert: (eventResult) => {

    module.exports._database.push(eventResult);
  },

  deleteAll: () => {

    let affected = module.exports._database.length;
    module.exports._database = [];
    return affected;
  },

  selectAll: () => {
    
    return module.exports._database.slice();
  }
}

File: CS.DotNetCore.LoadTest/src/CS.DotNetCore.LoadTest.WebApp/Data/Memory/EventResultMemoryDAO.cs
namespace CS.DotNetCore.LoadTest.WebApp.Data.Memory
{
    using Logging;
    using System.Collections.Concurrent;
    using System;
    using System.Linq;
    using System.Collections.Generic;

    internal class EventResultMemoryDAO : IEventResultDAO
    {
        private static readonly ConcurrentBag<EventResult> _dataBase = new ConcurrentBag<EventResult>();

        public void Insert(EventResult eventResult)
        {
            if (eventResult == null)
                throw new ArgumentNullException(nameof(eventResult));

            _dataBase.Add(eventResult);
        }

        public List<EventResult> SelectAll() {

            lock (_dataBase)
            {
                return _dataBase.ToList();
            }
        }

        public int DeleteAll()
        {
            lock (_dataBase)
            {
                var affected = _dataBase.Count;

                if (affected > 0)
                {
                    EventResult eventResult;

                    while (!_dataBase.IsEmpty)
                    {
                        _dataBase.TryTake(out eventResult);
                    }
                }

                return affected;
            }
        }
    }
}
